### PR TITLE
Gitignore

### PR DESCRIPTION
**Pull request by saecki**
This adds the entire `.idea` directory to the gitignore as Jetbrains themselves recommend. Because of this the code style xml files were removed so I added a `.editorconfig` to keep a consistent code style across the project and reformatted everything. The `.editorconfig` can be generated by IntelliJ if it doesn't seem right to you.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,557 @@
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+indent_style = space
+insert_final_newline = false
```
Comment (agersant) [inline .editorconfig]: Could you flip the setting to insert a new line at the end of files?

I gave up on tabs VS spaces long ago but this one I still care about.

**Review by agersant — CHANGES_REQUESTED** [review]
Excellent work once again!
The `.idea` noise had been bothering me for a while, but information on what is supposed to be versioned is very contradictory (and I didn't look into it very long). I'm glad you sorted it out!

Left one minor note for a personal preference in formatting.

File: .editorconfig (hunk 1 above)
Comment (saecki) [reply .editorconfig]: Done.

Comment (agersant) [reply .editorconfig]: Oh, I guess it does nothing for Java/Kotlin source files. Unfortunate.

**Pull request merged**